### PR TITLE
Prepare for packaging

### LINKED ISSUE #1: Add tests
Working on this now

### PR DESCRIPTION
**Pull request by scottcorgan**
Closes #1 

---

This is a bit of a large PR, but you can look at each commit to see its progress.

## What changed

* Sets it up as a CommonSJ module
* Adds tests to verify that it works
* Adds a WIP readme with instructions for viewing demo
* Adds example directory to play with (has live reload 👍 )
* Small refactor to pull the pure functions out of the main closure.

It's not ready to be published yet as we discuss #2. But a review and merge would be great 💯

### CODE REVIEW

**Review by shanebo — CHANGES_REQUESTED** [review]
Everything looks good! We don't need the yarn.lock though do we?

File: index.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      });
+    });
+
+    resizing = false;
+  }
+
+  window.addEventListener('resize', throttle);
+  resize();
+
+  // Tear down function
```
Comment (shanebo) [inline index.js:127]: @scottcorgan sincere question, why is this necessary in this case?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  function throttle() {
+    if (!resizing) {
+      window.requestAnimationFrame(resize);
+      resizing = true;
+    }
+  }
+
+  function resize() {
+    // eval(query.statment) uses these values for comparisons
+    const sw = window.innerWidth;
```
Comment (shanebo) [inline index.js:104]: @scottcorgan great catch pulling these out of the loop since they don't change on each iteration.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+  if (suiterClasses) {
+    classes = suiterClasses.split(',')
+      .map(function(style){
+        return cleanString(style).split(' ');
+      });
+  }
+
+  return queries
+    .map(cleanString)
```
Comment (shanebo) [inline index.js:68]: Good catch not needing to pass in a callback on line 68 and 69!

**Review by shanebo — CHANGES_REQUESTED** [review]

**Review by shanebo — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (scottcorgan) [reply index.js:127]: In a general use case, it probably doesn't need it, but it's a good idea to provide a way to let the consumer remove all remnants of the library without having to know which event handlers were attached.

In our codebase, this is will be because of Turbolinks. Also, it's helpful for testing 👍

**Comment by scottcorgan** [comment]
@shanebo  Appreciate you merging. I made some opinionated changes to the code, but I'm not totally married to them. Change things as you see fit.

- Moving things out of the closure is a personal preference and doesn't gain us anything in performance, etc.
- Using `setAttribute` instead of `dataset` as no gain for us except that JSDom doesn't support dataset yet. But I figured it was change that really doesn't affect the library use or api.
